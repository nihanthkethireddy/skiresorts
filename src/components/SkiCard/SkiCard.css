.card {
  height: 300px;
  width: 300px;
  background-color: red;
  margin: 10px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}