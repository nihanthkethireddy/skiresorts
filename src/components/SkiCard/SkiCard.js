import './SkiCard.css';

function SkiCard(props) {

  const {setPopup, deleteResort, resort} = props
  return <div className={'card'}>
    <div className={'resortDetails'} onClick={() => setPopup({open: true, resort: resort, readOnly: true})}>
      <img src={resort.img}></img>
      <span>Name: {resort.name}</span>
      <span>Ski Runs: {resort.runs}</span>
    </div>
    <div className={'resortActions'}>
      <button onClick={() => setPopup({open: true, resort: resort, readOnly: false})}>Edit</button>
      <button onClick={() => deleteResort(resort.id)}>Delete</button>
    </div>
  </div>
}

export {SkiCard};
