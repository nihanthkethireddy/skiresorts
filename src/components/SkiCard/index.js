import {SkiCard} from './SkiCard'

export {SkiCard}