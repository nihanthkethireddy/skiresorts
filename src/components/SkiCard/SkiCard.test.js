import { shallow } from 'enzyme';
import { SkiCard } from './SkiCard';

const props = {
  deleteResort: jest.fn(),
  setPopup: jest.fn(),
  resort: {
    img: null,
    name: 'Resort 1',
    runs: 100
  }
}
describe('<SkiCard /> with props', () => {
  const container = shallow(<SkiCard {...props} />);
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('Check if Edit and Delete buttons are rendered in SkiCard', () => {
    expect(container.find('button').length).toEqual(2);
  });

  it('Check if the image tag is rendered in SkiCard', () => {
    expect(container.find('img').length).toEqual(1);
  });

});