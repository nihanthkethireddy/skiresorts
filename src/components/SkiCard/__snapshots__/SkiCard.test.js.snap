// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkiCard /> with props should match the snapshot 1`] = `"<div class=\\"card\\"><div class=\\"resortDetails\\"><img/><span>Name: Resort 1</span><span>Ski Runs: 100</span></div><div class=\\"resortActions\\"><button>Edit</button><button>Delete</button></div></div>"`;
